lib/net/http/request.rb

```diff
@@ -1,8 +1,26 @@
 # frozen_string_literal: false
-# HTTP request class.
-# This class wraps together the request header and the request path.
-# You cannot use this class directly. Instead, you should use one of its
-# subclasses: Net::HTTP::Get, Net::HTTP::Post, Net::HTTP::Head.
+
+# This class is the base class for \Net::HTTP request classes;
+# it wraps together the request path and the request headers.
+#
+# The class should not be used directly;
+# instead you should use its subclasses:
+#
+# - \Net::HTTP::Get
+# - \Net::HTTP::Head
+# - \Net::HTTP::Post
+# - \Net::HTTP::Delete
+# - \Net::HTTP::Options
+# - \Net::HTTP::Trace
+# - \Net::HTTP::Patch
+# - \Net::HTTP::Put
+# - \Net::HTTP::Copy
+# - \Net::HTTP::Lock
+# - \Net::HTTP::Mkcol
+# - \Net::HTTP::Move
+# - \Net::HTTP::Propfind
+# - \Net::HTTP::Proppatch
+# - \Net::HTTP::Unlock
```

Comment:
How is this list sorted? Should we sort it alphabetically (or some other way) so it's more intuitive to read?

Reply:
They're in the order as given in the Wikipedia docs we've linked (which I assumed is the same as in the underlying specs):
- https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Request_methods
- https://en.wikipedia.org/wiki/WebDAV#Implementation
Want them alphabetical?

Reply:
I see, that makes sense. I wonder how Wikipedia orders their list 🤔